cogeo_mosaic/mosaic.py

```diff
@@ -0,0 +1,241 @@
+"""cogeo_mosaic.mosaic MosaicJSON models and helper functions."""
+
+from typing import Callable, Dict, List, Optional, Tuple, Sequence
+
+import warnings
+
+import click
+import mercantile
+from pygeos import STRtree, polygons, total_bounds
+from supermercado import burntiles
+
+from cogeo_mosaic.utils import get_footprints, _intersect_percent
+
+from pydantic import BaseModel, Field
+
+
+default_accessor: Callable = lambda feature: feature["properties"]["path"]
+
+
+class MosaicJSON(BaseModel):
+    """
+    MosaicJSON model.
+
+    Based on https://github.com/developmentseed/mosaicjson-spec
+
+    """
+
+    mosaicjson: str
+    name: Optional[str]
+    description: Optional[str]
+    version: str = "1.0.0"
+    attribution: Optional[str]
+    minzoom: int = Field(0, ge=0, le=30)
+    maxzoom: int = Field(30, ge=0, le=30)
+    quadkey_zoom: Optional[int]
+    bounds: List[float] = Field([-180, -90, 180, 90])
+    center: Optional[Tuple[float, float, int]]
+    tiles: Dict[str, List[str]]
+
+    @classmethod
+    def _filter(
+        cls,
+        tile: mercantile.Tile,
+        dataset: Sequence[Dict],
+        geoms: Sequence[polygons],
+        minimum_tile_cover=None,
+        tile_cover_sort=False,
+        maximum_items_per_tile: Optional[int] = None,
+    ):
+        """Filter and/or sort dataset per intersection coverage."""
+        indices = list(range(len(dataset)))
+
+        if minimum_tile_cover or tile_cover_sort:
+            tile_geom = polygons(mercantile.feature(tile)["geometry"]["coordinates"][0])
+            int_pcts = _intersect_percent(tile_geom, geoms)
+
+            if minimum_tile_cover:
+                indices = [ind for ind in indices if int_pcts[ind] > minimum_tile_cover]
+
+            if tile_cover_sort:
+                # https://stackoverflow.com/a/9764364
+                indices, _ = zip(*sorted(zip(indices, int_pcts), reverse=True))
```

Comment:
@kylebarron I added `reverse=True` here to match existent function

Reply:
Oh I forgot about that. Reverse sorts highest to lowest then?

Reply:
we reverse we sort % intersection from highest to lowest